public/scripts/controllers/AlbumsIndexController.js

```diff
@@ -0,0 +1,67 @@
+angular
+  .module('tunely')
+  .controller('AlbumsIndexController', AlbumsIndexController);
+
+AlbumsIndexController.$inject = ['$http'];
+
+function AlbumsIndexController ($http) {
+  var vm = this;
+  vm.newAlbum = {};
+  vm.newAlbum = {
+    name: 'Viva Hate',
+    artistName: 'Morrissey'
+  };
+
+  $http({
+    method: 'GET',
+    url: '/api/albums'
+  }).then(function successCallback(response) {
+    vm.albums = response.data;
+  }, function errorCallback(response) {
+    console.log('There was an error getting the data', response);
+  });
+
+  vm.createAlbum = function () {
+    $http({
+      method: 'POST',
+      url: '/api/albums',
+      data: vm.newAlbum,
+    }).then(function successCallback(response) {
+      vm.albums.push(response.data);
+    }, function errorCallback(response) {
+      console.log('There was an error posting the data', response);
+    });
+  }
+
+  vm.editAlbum = function (album) {
+    $http({
+      method: 'PUT',
+      url: '/api/albums/'+album._id,
+      data: album
+    }).then(function successCallback(json) {
+      // don't need to do anything!
+    }, function errorCallback(response) {
+      console.log('There was an error editing the data', response);
+    });
+  }
+
+  vm.deleteAlbum = function (id) {
+    $http({
+      method: 'DELETE',
+      url: '/api/albums/'+ id
+    }).then(function successCallback(json) {
+      var index = findWithAttr(vm.albums, '_id', id);
+      vm.albums.splice(index, 1);
+    }, function errorCallback(response) {
+      console.log('There was an error deleting the data', response);
+    });
+  }
+
+  function findWithAttr(array, attr, value) {
```

Comment:
This would take many of our students an entire drill session to work out right now.